mara_pipelines/pipelines.py

```diff
@@ -81,24 +81,49 @@ def html_doc_items(self) -> List[Tuple[str, str]]:
 
 
 class Task(Node):
-    def __init__(self, id: str, description: str, commands: Optional[List[Command]] = None, max_retries: Optional[int] = None) -> None:
+    def __init__(self, id: str, description: str, commands: Optional[Union[Callable, List[Command]]] = None, max_retries: Optional[int] = None) -> None:
         super().__init__(id, description)
-        self.commands = []
+        self.callable_commands = callable(commands)
```

Comment:
is_callable_commands? Something which indicates a boolean...

Reply:
Personally I would also simply save the state in `self._commands: None | list` vs `self._commands_callable: Callable[...] | None`-> one of them is `not None` depending if the argument is callable or not.

Reply:
Both true. I created now a property getter `is_dynamic_commands` and removed the variable. This will save memory :-) It is still good to have this available so that one can check upfront if the commands should be evaluated e.g. in the node_page.